modules/core/src/v2/coins/eth2.ts

```diff
@@ -0,0 +1,399 @@
+   * @param params
+   * @param params.userKey {String} [encrypted] xprv
+   * @param params.backupKey {String} [encrypted] xprv or xpub if the xprv is held by a KRS provider
+   * @param params.walletPassphrase {String} used to decrypt userKey and backupKey
+   * @param params.walletContractAddress {String} the ETH address of the wallet contract
+   * @param params.recoveryDestination {String} target address to send recovered funds to
+   * @param callback
+   */
+  recover(
+    params: RecoverOptions,
+    callback?: NodeCallback<RecoveryInfo | OfflineVaultTxInfo>
+  ): Bluebird<RecoveryInfo | OfflineVaultTxInfo> {
+    throw new Error('recover not implemented');
+  }
+
+  /**
+   * Make a query to Etherscan for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @param callback
+   * @returns {Object} response from Etherscan
+   */
+  recoveryBlockchainExplorerQuery(query: any, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      const response = yield request.get(common.Environments[self.bitgo.getEnv()].eth2ExplorerBaseUrl).query(query);
+
+      if (!response.ok) {
+        throw new Error('could not reach BeaconScan');
+      }
+      return response.body;
+    })
+      .call(this)
+      .asCallback(callback);
+  }
+
+  /**
+   * Fetch fee estimate information from the server
+   * @param {Object} params The params passed into the function
+   * @param {String} [params.recipient] The recipient of the transaction to estimate a send to
+   * @param {String} [params.data] The ETH tx data to estimate a send for
+   * @param callback
+   * @returns {Object} The fee info returned from the server
+   */
+  feeEstimate(params: FeeEstimateOptions, callback?: NodeCallback<FeeEstimate>): Bluebird<FeeEstimate> {
+    const self = this;
+    return co<FeeEstimate>(function* coFeeEstimate() {
+      const query: FeeEstimateOptions = {};
+      if (params && params.recipient) {
+        query.recipient = params.recipient;
+      }
+      if (params && params.data) {
+        query.data = params.data;
+      }
+      if (params && params.amount) {
+        query.amount = params.amount;
+      }
+
+      return self.bitgo
+        .get(self.url('/tx/fee'))
+        .query(query)
+        .result();
+    })
+      .call(this)
+      .asCallback(callback);
+  }
+
+  /**
+   * Generate BLS key pair
+   *
+   * @param seed - byte array to generate BLS key pair from
+   * @returns {Object} object with generated pub and prv
+   */
+  generateKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = new Eth2AccountLib.KeyPair();
+    if (seed) {
+      const prv = BLS.PrivateKey.fromBytes(seed).toHexString();
+      // @ts-ignore
+      keyPair.recordKeysFromPrivateKey(prv);
+    }
+
+    return {
+      // @ts-ignore
+      pub: keyPair.getKeys().publicKey.toHexString(),
```

Comment:
I guess I should have mentioned this on your account-lib PR, but personally I think that `keyPair.getKeys()` should already be returning the hex string serializations. Similar to reasoning above and below about abstracting the BLS library away. A user of the accountLib keyPair library shouldn't also have to concern themselves with this third party BLS library - we should just provide a generic interface to them